WordPress/Classes/ViewRelated/Plugins/PluginDirectoryViewModel.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import WordPressFlux
+import Gridicons
+
+protocol PluginListPresenter: class {
+    func present(site: JetpackSiteRef, query: PluginQuery)
+}
+
+class PluginDirectoryViewModel: Observable {
+
+    let site: JetpackSiteRef
+
+    let changeDispatcher = Dispatcher<Void>()
+
+    private let store: PluginStore
+    private var storeReceipt: Receipt?
+
+    private var installedReceipt: Receipt?
+    private var featuredReceipt: Receipt?
+    private var popularReceipt: Receipt?
+    private var newReceipt: Receipt?
+    private var searchReceipt: Receipt?
+
+    public init(site: JetpackSiteRef, store: PluginStore = StoreContainer.shared.plugin) {
+        self.store = store
+        self.site = site
+
+        installedReceipt = store.query(.all(site: site))
+        featuredReceipt = store.query(.featured(site: site))
+        popularReceipt = store.query(.feed(type: .popular))
+        newReceipt = store.query(.feed(type: .newest))
+
+        storeReceipt = store.onChange { [weak self] in
+            self?.changeDispatcher.dispatch()
+        }
+    }
+
+    func tableViewModel(presenter: PluginPresenter, listPresenter: PluginListPresenter) -> ImmuTable {
+
+        let installedRow = CollectionViewContainerRow<PluginDirectoryCollectionViewCell, Plugin>(
+            data: installedPlugins,
+            title: NSLocalizedString("Installed", comment: "Header of section in Plugin Directory showing installed plugins"),
+            secondaryTitle: NSLocalizedString("Manage", comment: "Button leading to a screen where users can manage their installed plugins"),
+            action: actionClosure(presenter: listPresenter, query: .all(site: site)),
+            configureCollectionCell: { [weak self] cell, plugin in
```

Comment:
Now that I see how `CollectionViewContainerRow` is being used, this feels out of place here. I understand it might be interesting to keep it generic, but then I'd have a factory type or method that deals with configuration.